packages/server/src/app/services/index.ts

```diff
@@ -0,0 +1,80 @@
+import { AdminServices } from './admin';
+import { AuthServices } from './auth';
+import { ContentServices } from './content';
+import { ImagesServices } from './images';
+import { SearchServices } from './search';
+
+export const InterfaceProviders = [
+    {
+        provide: 'IApprovalQueue',
+        useClass: AdminServices.find((x) => {
+            return x.name === 'ApprovalQueueService';
```

Comment:
hm. I like the way this is decoupled from things, but the hard reliance on a simple string is gonna be annoying when we want to rename things. I don't suppose TypeScript has something like C#'s `nameof(SomeClassHere)`? It'd be lovely if we could do something like `x.name === nameof(ApprovalQueueService)` (where `ApprovalQueueService` here is a _type_ rather than a string), and get automatic refactor support, instead of runtime explosion

Reply:
further exploration of this is required